TestFoundation/TestNSDateFormatter.swift

```diff
@@ -0,0 +1,178 @@
+            1469854638 : "Jul 30, 2016, 4:57:18 AM", 1470718638 : "Aug 9, 2016, 4:57:18 AM", 1473915438 : "Sep 15, 2016, 4:57:18 AM",
+            1477285038 : "Oct 24, 2016, 4:57:18 AM", 1478062638 : "Nov 2, 2016, 4:57:18 AM", 1482641838 : "Dec 25, 2016, 4:57:18 AM"
+        ]
+        
+        let f = NSDateFormatter()
+        f.dateStyle = .MediumStyle
+        f.timeStyle = .MediumStyle
+        f.timeZone = NSTimeZone(name: DEFAULT_TIMEZONE)
+        f.locale = NSLocale(localeIdentifier: DEFAULT_LOCALE)
+        
+        for (timestamp, stringResult) in timestamps {
+            
+            let testDate = NSDate(timeIntervalSince1970: timestamp)
+            let sf = f.stringFromDate(testDate)
+            
+            XCTAssertEqual(sf, stringResult)
+        }
+        
+    }
+    
+    
+    // LongStyle
+    // locale  stringFromDate  example
+    // ------  --------------  -----------------
+    // en_US   MMMM d, y       December 25, 2015
+    func test_dateStyleLong() {
+        
+        let timestamps = [
+            -31536000 : "January 1, 1969 at 12:00:00 AM GMT" , 0.0 : "January 1, 1970 at 12:00:00 AM GMT", 31536000 : "January 1, 1971 at 12:00:00 AM GMT",
+            2145916800 : "January 1, 2038 at 12:00:00 AM GMT", 1456272000 : "February 24, 2016 at 12:00:00 AM GMT", 1456358399 : "February 24, 2016 at 11:59:59 PM GMT",
+            1452574638 : "January 12, 2016 at 4:57:18 AM GMT", 1455685038 : "February 17, 2016 at 4:57:18 AM GMT", 1458622638 : "March 22, 2016 at 4:57:18 AM GMT",
+            1459745838 : "April 4, 2016 at 4:57:18 AM GMT", 1462597038 : "May 7, 2016 at 4:57:18 AM GMT", 1465534638 : "June 10, 2016 at 4:57:18 AM GMT",
+            1469854638 : "July 30, 2016 at 4:57:18 AM GMT", 1470718638 : "August 9, 2016 at 4:57:18 AM GMT", 1473915438 : "September 15, 2016 at 4:57:18 AM GMT",
+            1477285038 : "October 24, 2016 at 4:57:18 AM GMT", 1478062638 : "November 2, 2016 at 4:57:18 AM GMT", 1482641838 : "December 25, 2016 at 4:57:18 AM GMT"
+        ]
+        
+        let f = NSDateFormatter()
+        f.dateStyle = .LongStyle
+        f.timeStyle = .LongStyle
+        f.timeZone = NSTimeZone(name: DEFAULT_TIMEZONE)
+        f.locale = NSLocale(localeIdentifier: DEFAULT_LOCALE)
+        
+        for (timestamp, stringResult) in timestamps {
+            
+            let testDate = NSDate(timeIntervalSince1970: timestamp)
+            let sf = f.stringFromDate(testDate)
+            
+            XCTAssertEqual(sf, stringResult)
+        }
+        
+    }
+    
+    // FullStyle
+    // locale  stringFromDate  example
+    // ------  --------------  -------------------------
+    // en_US   EEEE, MMMM d, y  Friday, December 25, 2015
+    func test_dateStyleFull() {
+        
+        let timestamps = [
+            -31536000 : "Wednesday, January 1, 1969 at 12:00:00 AM GMT" , 0.0 : "Thursday, January 1, 1970 at 12:00:00 AM GMT",
```

Comment:
Shouldn't the correct value be `Greenwich Mean Time` instead of `GMT`?

That's what ICU spec says (http://userguide.icu-project.org/formatparse/datetime)
Also, that's the behaviour of `f.timeStyle = .full` on macOS.

> It appears that for negative timestampts (e.g. -31536000) DateFormatter on macOS wrongly returns GMT instead of Greenwich Mean Time, which seems to be a bug.